ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/filter/FilterManager.java

```diff
@@ -168,9 +167,9 @@ public void recordBlockEvent(final BlockAddedEvent event) {
         .filter(
             // Only keep filters where the "to" block could include the block in the event
             filter -> {
-              final OptionalLong maybeToBlockNumber = filter.getToBlock().getNumber();
+              final Optional<Long> maybeToBlockNumber = filter.getToBlock().getNumber();
               return maybeToBlockNumber.isEmpty()
-                  || maybeToBlockNumber.getAsLong() >= event.getBlock().getHeader().getNumber();
+                  || maybeToBlockNumber.get() >= event.getBlock().getHeader().getNumber();
```

Comment:
It's optional but I wonder if reducing this line like that would not be interesting. 
it seems to me to work

```java
filter(
     // Only keep filters where the "to" block could include the block in the event
     filter -> filter.getToBlock().getNumber().filter(number ->
              number>= event.getBlock().getHeader().getNumber()
     ).isPresent())
.forEach(...
```